assets/src/components/sharesecret/ShareSecretModal.tsx

```diff
@@ -0,0 +1,237 @@
+import {
+  Button,
+  Code,
+  FormField,
+  Input,
+  Modal,
+  ReturnIcon,
+  Toast,
+} from '@pluralsh/design-system'
+import React, {
+  Dispatch,
+  FormEvent,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import { useTheme } from 'styled-components'
+import { LayerPositionType } from '@pluralsh/design-system/dist/components/Layer'
+
+import { splitBindings } from '../settings/usermanagement/roles/RoleFormBindings'
+import { BindingInput } from '../utils/BindingInput'
+
+import {
+  PolicyBindingFragment,
+  useShareSecretMutation,
+} from '../../generated/graphql'
+import { isNonNullable } from '../../utils/isNonNullable'
+import { bindingToBindingAttributes } from '../settings/usermanagement/roles/misc'
+
+import { SECRETS_PATH } from '../../routes/secretsRoutesConsts'
+
+const getUrl = (handle?: string) =>
+  `https://${window.location.host}/${SECRETS_PATH}/${handle}`
+
+export default function ShareSecretModal({
+  open,
+  setOpen,
+}: {
+  open: boolean
+  setOpen: Dispatch<SetStateAction<boolean>>
+}) {
+  const theme = useTheme()
+  const [completed, setCompleted] = useState(false)
+  const [toast, setToast] = useState(false)
+  const [name, setName] = useState('')
+  const [secret, setSecret] = useState('')
+  const [bindings, setBindings] = useState<
+    Pick<PolicyBindingFragment, 'user' | 'group'>[]
+  >([])
+  const disabled = !name || !secret
+  const toastRef = useRef<HTMLElement>()
+
+  useEffect(() => {
```

Comment:
we should fix this in the DS at some point, generally want to avoid effects like this

Reply:
Totally agree.